aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1249,6 +1314,39 @@ private boolean s3KeyIdAccessKeyBothConfigured() {
     return (s3AccessKeyId == null) == (s3SecretAccessKey == null);
   }
 
+  private AwsCredentialsProvider credentialsProvider(String credentialsProviderClazz) {
```

Comment:
you mentioned 

> Most of the AWS SDK Credential Providers are using create() method instead of constructor

any reference for that? 

Because at least for the Trino case it's done by checking constructors. In Iceberg most uses `initialize(...)` instead of `create(...)`.

Reply:
I guess there is one case for create which is 

```
public static S3V4RestSignerClient create(Map<String, String> properties) {
    return ImmutableS3V4RestSignerClient.builder().properties(properties).build();
  }
```

in that case I am fine with this. But I think we should still support only 1 method for `create(Map)` instead of an additional `create()` without argument.

Reply:
In this case I would probably first look for a "map" create method and pass an empty map if we have no properties. If we don't have no-arg create method we can fall back to that if the first method doesn't exist. I think your filter code above guarantees that properties is non-null but we can check that. 

@jackye1995 knows more about what credential providers there are, but I'm surprised that having a constructor is that rare. Anyway I definitely think we should always call a "create(map) if that exists"

```java
try {
  getCreate(Map)
  invokeMapCreate(properties)
} catch (NoSuchMethod) {
  try {
     getCreate()
     invokeCreate()
   }
}
```

Reply:
Thanks @RussellSpitzer .  I am fine with your approach. Once @jackye1995 approves, I can add it.

hi @jackye1995 These are some of the AWS SDK provided credentials provider classes

```
software.amazon.awssdk.auth.credentials.AnonymousCredentialsProvider
software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider
software.amazon.awssdk.auth.credentials.SystemPropertyCredentialsProvider
software.amazon.awssdk.auth.credentials.WebIdentityTokenFileCredentialsProvider
```

AWS SDK V2 Apis not using constructors in most of the cases.

Reply:
@ru